shunting-yard-replicator/src/main/java/com/hotels/shunting/yard/replicator/exec/receiver/ContextFactory.java

```diff
@@ -107,13 +115,26 @@ public Context createContext(MetaStoreEvent event) {
 
   private CircusTrainConfig generateConfiguration(MetaStoreEvent event) {
     String sourceMetaStoreUri = getSourceMetaStoreUri(event);
-    String replicaTableLocation = workOutReplicaLocation(event.getDatabaseName(), event.getTableName());
+
+    String replicaDatabaseName = event.getDatabaseName();
+    String replicaTableName = event.getTableName();
+
+    TableReplication tableReplication = shuntingYardTableReplications
+        .getTableReplicationsMap()
+        .get(String.join(".", event.getDatabaseName(), event.getTableName()));
+
+    if (tableReplication != null) {
+      replicaDatabaseName = tableReplication.getReplicaDatabaseName();
```

Comment:
Is the null check needed here? It's already checked in `ShuntingYardTableReplications` when the `tableReplications` are added to the map

Reply:
Ah I see, if its null we want to use the event to work out the location instead. Never mind! :)